src/WidgetBase.ts

```diff
@@ -328,41 +292,57 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 	public __setProperties__(originalProperties: P): void {
 		const { bind, ...properties } = originalProperties as any;
 		const changedPropertyKeys: string[] = [];
-		const allProperties = new Set([...Object.keys(properties), ...Object.keys(this._properties)]);
+		const allProperties = [ ...Object.keys(properties), ...Object.keys(this._properties) ];
+		const checkedProperties: string[] = [];
 		const diffPropertyResults: any = {};
+		const registeredDiffPropertyNames = this.getDecorator('registeredDiffProperty');
+		let runReactions = false;
 
 		this._renderState = WidgetRenderState.PROPERTIES;
-		this._bindFunctionProperties(properties, bind);
 
-		allProperties.forEach((propertyName) => {
+		for (let i = 0; i < allProperties.length; i++) {
+			const propertyName = allProperties[i];
+			if (checkedProperties.indexOf(propertyName) > 0) {
+				continue;
+			}
+			checkedProperties.push(propertyName);
 			const previousProperty = this._properties[propertyName];
-			const newProperty = properties[propertyName];
-			const diffFunctions: DiffPropertyFunction[] = this.getDecorator(`diffProperty:${propertyName}`);
-
-			if (diffFunctions.length === 0) {
-				diffFunctions.push(auto);
+			const newProperty = this._bindFunctionProperty(properties[propertyName], bind);
+			if (registeredDiffPropertyNames.indexOf(propertyName) !== -1) {
```

Comment:
I "think" that `indexOf` performs better across our supported platforms, although native `includes` is slightly better on chrome and firefox according to this benchmark: https://jsperf.com/array-indexof-vs-includes